editor/components/inserter/menu.js

```diff
@@ -12,11 +13,59 @@ import Dashicon from 'components/dashicon';
 class InserterMenu extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
+		this.blockTypes = wp.blocks.getBlocks();
+		this.categories = wp.blocks.getCategories();
```

Comment:
That was the assumption I was making.  I would advise against ever having these values change, and instead use filtering to get the changed values.